rotkehlchen/chain/ethereum/decoding/liquity/decoder.py

```diff
@@ -0,0 +1,106 @@
+
+from rotkehlchen.accounting.structures.base import (
+    HistoryBaseEntry,
+    HistoryEventSubType,
+    HistoryEventType,
+    get_tx_event_type_identifier,
+)
+from rotkehlchen.chain.ethereum.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.ethereum.decoding.structures import ActionItem, TxEventSettings
+from rotkehlchen.chain.ethereum.structures import EthereumTxReceiptLog
+from rotkehlchen.chain.ethereum.types import string_to_ethereum_address
+from rotkehlchen.constants.assets import A_ETH, A_LUSD
+from rotkehlchen.types import ChecksumEthAddress, EthereumTransaction
+
+if TYPE_CHECKING:
+    from rotkehlchen.accounting.pot import AccountingPot
+
+BALANCE_UPDATE = b'\xca#+Z\xbb\x98\x8cT\x0b\x95\x9f\xf6\xc3\xbf\xae>\x97\xff\xf9d\xfd\t\x8cP\x8f\x96\x13\xc0\xa6\xbf\x1a\x80'  # noqa: E501
+STABILITY_POOL = string_to_ethereum_address('0xDf9Eb223bAFBE5c5271415C75aeCD68C21fE3D7F')
+
+CPT_LIQUITY = 'liquity'
+
+
+class LiquityDecoder(DecoderInterface):  # lgtm[py/missing-call-to-init]
+
+    def _decode_trove_operations(  # pylint: disable=no-self-use
+            self,
+            tx_log: EthereumTxReceiptLog,
+            transaction: EthereumTransaction,  # pylint: disable=unused-argument
+            decoded_events: List[HistoryBaseEntry],
+            all_logs: List[EthereumTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: List[ActionItem],  # pylint: disable=unused-argument
+    ) -> Tuple[Optional[HistoryBaseEntry], Optional[ActionItem]]:
+        if tx_log.topics[0] != BALANCE_UPDATE:
+            return None, None
+
+        for event in decoded_events:
+            if event.event_type == HistoryEventType.RECEIVE and event.asset == A_LUSD:
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.GENERATE_DEBT
+                event.counterparty = CPT_LIQUITY
+                event.notes = f'Generate {event.balance.amount} {A_LUSD.symbol} from liquity'
+            elif event.event_type == HistoryEventType.SPEND and event.asset == A_LUSD:
+                event.event_type = HistoryEventType.SPEND
+                event.event_subtype = HistoryEventSubType.PAYBACK_DEBT
+                event.counterparty = CPT_LIQUITY
+                event.notes = f'Return {event.balance.amount} {A_LUSD.symbol} to liquity'
+            elif event.event_type == HistoryEventType.SPEND and event.event_subtype == HistoryEventSubType.NONE and event.asset == A_ETH:  # noqa: E501
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.counterparty = CPT_LIQUITY
+                event.notes = f'Deposit {event.balance.amount} {event.asset.symbol} as collateral for liquity'  # noqa: E501
+            elif event.event_type == HistoryEventType.RECEIVE and event.event_subtype == HistoryEventSubType.NONE and event.asset == A_ETH:  # noqa: E501
```

Comment:
Is ETH the only collateral type in Liquity?

Reply:
yes

Reply:
Ah that's cool and easy then.